sandbox/modules/identity/src/test/java/org/opensearch/identity/rest/action/RestCreateUserActionTests.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.action;
+
+import org.apache.lucene.util.SetOnce;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionResponse;
+import org.opensearch.action.ActionType;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.bytes.BytesArray;
+import org.opensearch.common.collect.Map;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.identity.rest.request.CreateUserRequest;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.client.NoOpNodeClient;
+import org.opensearch.test.rest.FakeRestChannel;
+import org.opensearch.test.rest.FakeRestRequest;
+
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Tests to verify the behavior of rest create user action
+ */
+public class RestCreateUserActionTests extends OpenSearchTestCase {
+    public void testParseCreateUserRequestWithInvalidJsonThrowsException() {
+        RestCreateUserAction action = new RestCreateUserAction();
+        RestRequest request = new FakeRestRequest.Builder(xContentRegistry()).withContent(
+            new BytesArray("{invalid_json}"),
+            XContentType.JSON
+        ).build();
+        Exception e = expectThrows(IllegalArgumentException.class, () -> action.prepareRequest(request, null));
+        assertThat(e.getMessage(), equalTo("Failed to parse request body"));
```

Comment:
I know that this is the exception message that is set above. What do you think about making it a bit more specific like "Failed to parse user create request" or something along those lines?

Reply:
sounds good